core/src/either.rs

```diff
@@ -385,11 +343,12 @@ impl<A: ProtocolName, B: ProtocolName> ProtocolName for EitherName<A, B> {
         }
     }
 }
-
+#[pin_project(project = EitherTransportProj)]
 #[derive(Debug, Copy, Clone)]
+#[must_use = "futures do nothing unless polled"]
```

Comment:
```suggestion
#[must_use = "transports do nothing unless polled"]
```

Not quite a `Future` but rather a `Transport`, right?